Userland/Applications/SoundPlayer/M3UParser.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2021, Cesar Torres <shortanemoia@protonmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "M3UParser.h"
+#include <AK/NonnullOwnPtrVector.h>
+#include <AK/OwnPtr.h>
+#include <AK/RefPtr.h>
+#include <AK/Utf8View.h>
+#include <LibCore/FileStream.h>
+#include <ctype.h>
+
+M3UParser::M3UParser()
+{
+}
+
+NonnullOwnPtr<M3UParser> M3UParser::from_file(const String path)
+{
+
+    auto parser = make<M3UParser>();
+    VERIFY(!path.is_null() && !path.is_empty() && !path.is_whitespace());
+    parser->m_use_utf8 = path.ends_with(".m3u8", AK::CaseSensitivity::CaseInsensitive);
+    FILE* file = fopen(path.characters(), "r");
+    VERIFY(file != nullptr);
+    fseek(file, 0, SEEK_END);
+    size_t file_size = ftell(file);
+    fseek(file, 0, SEEK_SET);
+    VERIFY(file_size > 0);
+    Vector<u8> temp_buffer;
+    temp_buffer.resize(file_size);
+    auto bytes_read = fread(temp_buffer.data(), 1, file_size, file);
+    VERIFY(bytes_read == file_size);
+    parser->m_m3u_raw_data = *new String(temp_buffer.span(), NoChomp);
```

Comment:
I _feel_ like you should be able to use [a `Core::File` object](https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibCore/File.h) here, but it doesn't seem like it has an easy `read_all_to_string() -> String` method...
Maybe there's something that could be done with [`Core::File::read_all() -> ByteBuffer`](https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibCore/IODevice.h#L84) ?

Reply:
Good point, it makes the code shorter too.